astropy/coordinates/representation.py

```diff
@@ -1990,10 +1987,10 @@ class BaseDifferential(BaseRepresentationOrDifferential):
 
     @classmethod
     def _check_base(cls, base):
-        if not isinstance(base, cls.base_representation):
-            raise TypeError('need a base of the correct representation type, '
-                            '{0}, not {1}.'.format(cls.base_representation,
-                                                   type(base)))
+        if cls not in base._compatible_differentials:
```

Comment:
Looking at this again, it is slightly odd to me that `base` has to know which differentials are compatible; it seems more logical for the differentials to know which works to transform them (after all, they have the logic to do that in `represent_as`, etc.). See #6287 with this and possible other "implementation details" that we might consider fixing.